File: XText18.cpp
#include "pt4.h"
#include <fstream>
using namespace std;

void Solve()
{
    Task("XText18");
    int k;
    string s1,s2 = "tmp.txt";
    pt >> k >> s1;
    ifstream f1(s1, ios::binary);
	ofstream f2(s2, ios::binary);
	for(int i = 0; i < k; i ++){
		char c;
		f1.read((char *)&c, sizeof(c));
		if(c == '\n'){
			f2.write((char *)&c, sizeof(c));
			//f1.seekg(k, ios::cur);
			f1.seekg(k, ios::cur);
			f1.seekg(f1.tellg());
			break;
		}
	}
	while (f1.peek() != -1)
	{
		char c;
		f1.read((char *)&c, sizeof(c));
		f2.write((char *)&c, sizeof(c));
		if(c == '\n'){
			
			for(int i = 0; i < k; i ++){
				f1.read((char *)&c, sizeof(c));
				
				if(c == '\n' && f1.peek() != -1){
					f2.write((char *)&c, sizeof(c));
					f1.seekg(k, ios::cur);
					f1.seekg(f1.tellg());
					break;
				}

				
			}
			
		}
	}
	f1.close();
	f2.close();
	remove(s1.c_str());
	rename(s2.c_str(), s1.c_str());
}

File: XText18.cs
// File: "XText18"
using PT4;
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XText18");
            int k = GetInt();
            string s1 = GetString(), s2 = "tmp.txt";
            StreamReader rf = new StreamReader(s1);
            StreamWriter wf = new StreamWriter(s2);
            while (rf.Peek() != -1)
            {
                string line = string.Empty;
                line = rf.ReadLine();
                if (line.Length <= k)
                {
                    line = "";
                    wf.WriteLine(line);
                }
                else
                {
                    line = line.Remove(0,k);
                    wf.WriteLine(line);
                }
            }
            rf.Close();
            wf.Close();
            File.Delete(s1);
            File.Move(s2, s1);
        }
    }
}
